### PR TITLE
feat(Livestream): Add samples and tests

### PR DESCRIPTION
**Pull request by irataxy**
b:241866780

Part 2 of 3: Channels. See also https://cloud.google.com/livestream/docs/how-to/create-channels

### CODE REVIEW

File: media/livestream/api/LiveStream.Samples.Tests/LiveStreamFixture.cs [hunk 1]
```diff
@@ -50,26 +65,99 @@ public LiveStreamFixture()
...
+        InputIds.Add(TestUpdateInputId);
+        TestUpdateInput = await _createInputSample.CreateInputAsync(ProjectId, LocationId, TestUpdateInputId);
+
+        TestChannelId = $"{ChannelIdPrefix}-{RandomId()}";
+        ChannelIds.Add(TestChannelId);
+        TestChannel = await _createChannelSample.CreateChannelAsync(ProjectId, LocationId, TestChannelId, TestInputId, ChannelOutputUri);
     }
 
-    public async Task DisposeAsync()
+    public async Task CleanOutdatedResources()
```
Comment (amanda-tarafa) [inline media/livestream/api/LiveStream.Samples.Tests/LiveStreamFixture.cs:83]: I'd like to know why were this channels/inputs left behind in the first place. You are already deleting all channels/inputs you created on Dispose so this should never have anything to delete, right?

File: media/livestream/api/LiveStream.Samples.Tests/UpdateChannelAsyncTests.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        public UpdateChannelAsyncTest(LiveStreamFixture fixture)
+        {
+            _fixture = fixture;
+            _updateSample = new UpdateChannelSample();
+        }
+
+        [Fact]
+        public async Task UpdatesChannelAsync()
+        {
+            var result = await _updateSample.UpdateChannelAsync(_fixture.ProjectId, _fixture.LocationId, _fixture.TestChannelId, _fixture.TestUpdateInputId);
```
Comment (amanda-tarafa) [inline media/livestream/api/LiveStream.Samples.Tests/UpdateChannelAsyncTests.cs:37]: Does updating the TestChannelId has side effects that may affect other tests using the same channel? If yes, then maybe you need to create a channel specifically for updating.

File: media/livestream/api/LiveStream.Samples.Tests/StopChannelAsyncTests.cs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            _startChannelSample = new StartChannelSample();
+            _stopChannelSample = new StopChannelSample();
+            _getSample = new GetChannelSample();
+        }
+
+        public async Task InitializeAsync()
+        {
+            await _createInputSample.CreateInputAsync(
+                    _fixture.ProjectId, _fixture.LocationId,
+                    _inputId);
```
Comment (amanda-tarafa) [inline media/livestream/api/LiveStream.Samples.Tests/StopChannelAsyncTests.cs:55]: nit:
```suggestion
            await _createInputSample.CreateInputAsync(
                    _fixture.ProjectId, _fixture.LocationId, _inputId);
```

File: media/livestream/api/LiveStream.Samples.Tests/StartChannelAsyncTests.cs [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        }
+
+        public async Task InitializeAsync()
+        {
+            await _createInputSample.CreateInputAsync(
+                    _fixture.ProjectId, _fixture.LocationId,
+                    _inputId);
+
+            await _createChannelSample.CreateChannelAsync(
+        _fixture.ProjectId, _fixture.LocationId, _channelId, _inputId, _fixture.ChannelOutputUri);
```
Comment (amanda-tarafa) [inline media/livestream/api/LiveStream.Samples.Tests/StartChannelAsyncTests.cs:56]: Check these line breaks and indents everywhere.

```suggestion
            await _createInputSample.CreateInputAsync(
                    _fixture.ProjectId, _fixture.LocationId, _inputId);

            await _createChannelSample.CreateChannelAsync(
                _fixture.ProjectId, _fixture.LocationId, _channelId, _inputId, _fixture.ChannelOutputUri);
```

File: media/livestream/api/LiveStream.Samples.Tests/DeleteChannelAsyncTests.cs [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        }
+
+        public async Task InitializeAsync()
+        {
+            await _createInputSample.CreateInputAsync(
+                    _fixture.ProjectId, _fixture.LocationId,
+                    _inputId);
+
+            await _createChannelSample.CreateChannelAsync(
+        _fixture.ProjectId, _fixture.LocationId, _channelId, _inputId, _fixture.ChannelOutputUri);
```
Comment (amanda-tarafa) [inline media/livestream/api/LiveStream.Samples.Tests/DeleteChannelAsyncTests.cs:53]:
```suggestion
               _fixture.ProjectId, _fixture.LocationId, _channelId, _inputId, _fixture.ChannelOutputUri);
```

**Review by amanda-tarafa — CHANGES_REQUESTED** [review]
Looking good overall, just a few indentation and line break nits.
And a couple of questions, we can chat about these later.

Also, I've fixed CI so now you get the logs in the PR.

File: media/livestream/api/LiveStream.Samples.Tests/LiveStreamFixture.cs (hunk 1 above)
Comment (irataxy) [reply media/livestream/api/LiveStream.Samples.Tests/LiveStreamFixture.cs:83]: Correct. Possibly the entire test timed out, leaving the channels and inputs.

File: media/livestream/api/LiveStream.Samples.Tests/UpdateChannelAsyncTests.cs (hunk 2 above)
Comment (irataxy) [reply media/livestream/api/LiveStream.Samples.Tests/UpdateChannelAsyncTests.cs:37]: The TestChannel is only used by GetChannel, ListChannels, and UpdateChannel, so I don't believe there are side effects.

**Review by amanda-tarafa — APPROVED** [review]
LGTM, thanks!

**Pull request merged**